### ISSUE TITLE
Bazel 0.19.1 fails: Toolchain identifier 'local' was not found, valid identifiers are [local_linux, local_darwin, local_windows]

### ISSUE THREAD

**Issue by byronyi** [tensorflow/tensorflow#23784]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 16.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): source
- TensorFlow version: 2226a4d599224d4759844db5c80460fafd87145f
- Python version: 2.7.12
- Installed using virtualenv? pip? conda?: virtualenv
- Bazel version (if compiling from source): 0.19.1
- GCC/Compiler version (if compiling from source): gcc (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0
- CUDA/cuDNN version: 9.0/7.2.1.38-1+cuda9.0
- GPU model and memory: V100-PCIE-32GB

**Describe the problem**

```
$ bazel clean
Starting local Bazel server and connecting to it...
INFO: Starting clean (this may take a while). Consider using --async if the clean takes more than several minutes.
$ ./configure
WARNING: Running Bazel server needs to be killed, because the startup options are different.
WARNING: --batch mode is deprecated. Please instead explicitly shut down your Bazel server using the command "bazel shutdown".
You have bazel 0.19.1 installed.
Please specify the location of python. [Default is /home/byronyi/tf/bin/python]:


Traceback (most recent call last):
  File "<string>", line 1, in <module>
AttributeError: 'module' object has no attribute 'getsitepackages'
Found possible Python library paths:
  /home/byronyi/tf/lib/python2.7/site-packages
Please input the desired Python library path to use.  Default is [/home/byronyi/tf/lib/python2.7/site-packages]

Do you wish to build TensorFlow with XLA JIT support? [Y/n]:
XLA JIT support will be enabled for TensorFlow.

Do you wish to build TensorFlow with OpenCL SYCL support? [y/N]:
No OpenCL SYCL support will be enabled for TensorFlow.

Do you wish to build TensorFlow with ROCm support? [y/N]:
No ROCm support will be enabled for TensorFlow.

Do you wish to build TensorFlow with CUDA support? [y/N]: y
CUDA support will be enabled for TensorFlow.

Please specify the CUDA SDK version you want to use. [Leave empty to default to CUDA 9.0]:


Please specify the location where CUDA 9.0 toolkit is installed. Refer to README.md for more details. [Default is /usr/local/cuda]:


Please specify the cuDNN version you want to use. [Leave empty to default to cuDNN 7]:


Please specify the location where cuDNN 7 library is installed. Refer to README.md for more details. [Default is /usr/local/cuda]:


Do you wish to build TensorFlow with TensorRT support? [y/N]:
No TensorRT support will be enabled for TensorFlow.

Please specify the locally installed NCCL version you want to use. [Default is to use https://github.com/nvidia/nccl]:


Please specify a list of comma-separated Cuda compute capabilities you want to build with.
You can find the compute capability of your device at: https://developer.nvidia.com/cuda-gpus.
Please note that each additional compute capability significantly increases your build time and binary size. [Default is: 3.5,7.0]: 7.0


Do you want to use clang as CUDA compiler? [y/N]:
nvcc will be used as CUDA compiler.

Please specify which gcc should be used by nvcc as the host compiler. [Default is /usr/bin/gcc]:


Do you wish to build TensorFlow with MPI support? [y/N]:
No MPI support will be enabled for TensorFlow.

Please specify optimization flags to use during compilation when bazel option "--config=opt" is specified [Default is -march=native -Wno-sign-compare]:


Would you like to interactively configure ./WORKSPACE for Android builds? [y/N]:
Not configuring the WORKSPACE for Android builds.

Preconfigured Bazel build configs. You can use any of the below by adding "--config=<>" to your build command. See .bazelrc for more details.
	--config=mkl         	# Build with MKL support.
	--config=monolithic  	# Config for mostly static monolithic build.
	--config=gdr         	# Build with GDR support.
	--config=verbs       	# Build with libverbs support.
	--config=ngraph      	# Build with Intel nGraph support.
	--config=dynamic_kernels	# (Experimental) Build kernels into separate shared objects.
Preconfigured Bazel build configs to DISABLE default on features:
	--config=noaws       	# Disable AWS S3 filesystem support.
	--config=nogcp       	# Disable GCP support.
	--config=nohdfs      	# Disable HDFS support.
	--config=noignite    	# Disable Apacha Ignite support.
	--config=nokafka     	# Disable Apache Kafka support.
Configuration finished
$ bazel build -c opt --config=opt //tensorflow/tools/pip_package:build_pip_package
Starting local Bazel server and connecting to it...
ERROR: /home/byronyi/.cache/bazel/_bazel_byronyi/fe336f47afec8b033fb4b29a6628548f/external/local_config_cc/BUILD:57:1: in cc_toolchain rule @local_config_cc//:cc-compiler-k8: Error while selecting cc_toolchain: Toolchain identifier 'local' was not found, valid identifiers are [local_linux, local_darwin, local_windows]
ERROR: Analysis of target '//tensorflow/tools/pip_package:build_pip_package' failed; build aborted: Analysis of target '@local_config_cc//:cc-compiler-k8' failed; build aborted
INFO: Elapsed time: 2.701s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (228 packages loaded, 1224 targets configured)
    currently loading: tensorflow/core ... (6 packages)
```

**Any other info / logs**
Manually changing 

```/home/byronyi/.cache/bazel/_bazel_byronyi/fe336f47afec8b033fb4b29a6628548f/external/local_config_cc/BUILD:57``` 

from `local` to `local_linux` fix the problem.

**Comment by byronyi** [comment]
#23719 seems to be same issue on Windows with Bazel 0.19.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @wt-huang: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by byronyi** [comment]
This is fixed in master. Closing for now.

**Comment by cbasavaraj** [comment]
Changing to `local_linux works` with Bazel 0.19.1
I had to come back to this version from Bazel 0.20 as there was another issue with that:
`bazel build issue (does not contain a toolchain for CPU 'k8')`

And of course can't do a pip install because CUDA 9.1 is not supported. And let's not even talk about Python 3.7 compatibility issues.

Seriously, if any of the (no doubt good) folks at TF are reading this, the barriers to entry / using TensorFlow regularly are so high and painful compared to PyTorch, it's a big turn off.

I use PyTorch for most of my work, but occasionally I find a repo only in TF and am forced to use it. I regret it every time.

**Comment by rohit-gupta** [comment]
@cbasavaraj Hi, in which file do I need to make this fix ? I am facing the same issue and when I grep "local" in the directory it appears in hundreds of files

**Comment by cbasavaraj** [comment]
@rohit-gupta See @byronyi 's original post: 
`/home/byronyi/.cache/bazel/_bazel_byronyi/fe336f47afec8b033fb4b29a6628548f/external/local_config_cc/BUILD:57 `
It will be the equivalent location for you

**Comment by Bortrex** [comment]
> Changing to `local_linux works` with Bazel 0.19.1
> I had to come back to this version from Bazel 0.20 as there was another issue with that:
> `bazel build issue (does not contain a toolchain for CPU 'k8')`
> 
> And of course can't do a pip install because CUDA 9.1 is not supported. And let's not even talk about Python 3.7 compatibility issues.
> 
> Seriously, if any of the (no doubt good) folks at TF are reading this, the barriers to entry / using TensorFlow regularly are so high and painful compared to PyTorch, it's a big turn off.
> 
> I use PyTorch for most of my work, but occasionally I find a repo only in TF and am forced to use it. I regret it every time.

thanks Dude! And true, using/installing pytorch is way to easy comparing with TF... :/

**Comment by edumucelli** [comment]
@cbasavaraj that is so true, same issues, not support to CUDA 9.1, everytime is a struggle with Bazel such as [#24385](https://github.com/tensorflow/tensorflow/issues/24385). There is a great comment in that linked thread. "[...] Tensorflow stuffs Bazel down our throat. If CMake was a first class citizen a lot of us wouldn't touch bazel with a 10foot pole :)"

**Comment by cbasavaraj** [comment]
Hey guys, I installed tf-gpu once recently. Was pleasantly surprised to find the installation went smoothly. I followed the instructions here for the https://www.tensorflow.org/install/gpu See in particular https://www.tensorflow.org/install/gpu#ubuntu_1804_cuda_10 for the CUDA etc depedencies. 

I still use PyTorch for 95% of my work though. There are a couple of pretrained nets I use tf for, and find that it throws a lot of warnings and messes up my command line logs! That apart, I haven't myself coded up / trained a TF model recently, so I don't know if it's become any easier that it used to be. I guess TF 2.0 will be pretty good.
